src/utils/embedding_store.py

```
@@ -293,6 +293,66 @@
+                )
+                return f"{query_prefix}{query}{context_prefix}{truncated_context}"
+
+        # Query itself is too long - truncate it
+        available_query_tokens = max_tokens - prefix_tokens
+        if available_query_tokens > 0:
+            # Keep the beginning of the query (more important than the end)
+            truncated_query = encoding.decode(query_tokens[:available_query_tokens])
+            return f"{query_prefix}{truncated_query}"
+
+        # Pathological case - just return what we can
+        logger.warning(f"Token limit too restrictive: {max_tokens}")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)